apps/dashr-yield-curve/app.R

```diff
@@ -0,0 +1,428 @@
+library(plotly)
+library(dplyr)
+library(tidyr)
+library(data.table)
+library(purrr)
+library(quantmod)
+library(magrittr)
+library(dash)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+library(glue)
+library(jsonlite)
+library(rlang)
+
+#setwd("~/Dash apps/dashr-yield-curve")
+
+appName <- Sys.getenv("DASH_APP_NAME")
+pathPrefix <- sprintf("/%s/", appName)
+
+
+Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+           DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+setwd(sprintf("/app/apps/%s", appName))
+
+l. <- list
+
+#Helper functions
+
+genMarksEmpty <- function(n){
+  l <- list(glue(''))
+  names(l) <- 'label'
+  m <- list(l)
+  names(m) <- glue('{n}')
+  return(m)
+}
+
+safemerge <- function(v1, v2){
+  #for columns for vectors use mapply(safemerge, v1, v2)
+  if(is.na(v1)||is.null(v1)){
+    return(v2)
+  } else{
+    return(v1)
+  }
+  
+}
+
+
+safelabel <- function(v1){
+  
+  if(is.na(v1)){
+    return("3-month")
+  } else{
+    return('1-month')
+  }
+  
+}
+
+###############INTERNAL LOGIC######################
+
+last_back <- 0
+last_next <- 0
+
+DT_ <- fread('data/yield_curve.csv') %>% drop_na(., x,y)
+dates <- DT_$y
+Dates <- base::as.Date(dates, format='%m/%d/%Y')
+labels <-(DT_$x)[1:11]
+DT <- DT_[, 3:13]
+colnames(DT) <- labels
+rownames(DT) <- dates
+DM <- DT %>% data.matrix()
+
+UPS <- l.('0'= l.(x=0, y=0, z=1), '1'=l.(x=0, y=0, z=1), '2'=l.(x=0, y=0, z=1), '3'=l.(x=0, y=0, z=1), '4'=l.(x=0, y=0, z=1), '5'=l.(x=0, y=0, z=1))
+
```

Comment:
I really suggest to replace all `l.` to `list` to make the code more like R instead of python